python_modules/libraries/dagster-sigma/dagster_sigma/resource.py

```diff
@@ -0,0 +1,65 @@
+from enum import Enum
+from typing import Any, Dict, Optional
+
+import requests
+from dagster import ConfigurableResource
+from pydantic import Field, PrivateAttr
+
+
+class SigmaBaseUrl(str, Enum):
+    AWS_US = "https://aws-api.sigmacomputing.com"
+    AWS_CANADA = "https://api.ca.aws.sigmacomputing.com"
+    AWS_EUROPE = "https://api.eu.aws.sigmacomputing.com"
+    AWS_UK = "https://api.uk.aws.sigmacomputing.com"
+    AZURE_US = "https://api.us.azure.sigmacomputing.com"
+    GCP = "https://api.sigmacomputing.com"
+
+
+class SigmaOrganization(ConfigurableResource):
+    """Represents a workspace in PowerBI and provides utilities
+    to interact with the PowerBI API.
+    """
+
+    base_url: str = Field(
+        ...,
+        description=(
+            "Base URL for the cloud type of your Sigma organization, found under the Administration -> Account -> Site settings."
+            " See https://help.sigmacomputing.com/docs/region-warehouse-and-feature-support for more information."
```

Comment:
https://help.sigmacomputing.com/reference/get-started-sigma-api#identify-your-api-request-url is probably a better URL